src/mqt/predictor/rl/helper.py

```diff
@@ -199,6 +201,16 @@ def get_actions_opt() -> list[dict[str, Any]]:
     ]
 
 
+def get_actions_opt_before_layout() -> list[dict[str, Any]]:
+    return [
+        {
+            "name": "BQSKitO2",
+            "transpile_pass": lambda circuit: bqskit_compile(circuit, optimization_level=2),
+            "origin": "bqskit",
+        },
+    ]
```

Comment:
Overall this looks good to me.
However, as also indicated by your offline message, we probably only want to apply this action whenever no layout is set.
On the other hand, BQSKit's MachineModel also takes a coupling map as an argument. So, technically, one could apply a very similar reasoning to how we handle TKET at the moment.
So if no layout is set, no coupling map is passed. If a layout is set, the coupling map is passed in as well.
Naturally, this also requires handling the layout properly.
Technicall, you could make this check a little easier by moving the Qiskit->BQSKit translation into the action here.

Reply:
I have added a BQSKit synthesize and mapping action, additionally I handcrafted the respective layout as well. It would be great, if you could check the new setup. It was a bit tedious to implement it and a second person checking it would be helpful. Thanks :)